social-meli/src/test/java/com/api/social_meli/unit/service/UserServiceTest.java

```diff
@@ -1,4 +1,82 @@
 package com.api.social_meli.unit.service;
 
+import com.api.social_meli.dto.MessageDto;
+import com.api.social_meli.exception.NotFoundException;
+import com.api.social_meli.model.User;
+import com.api.social_meli.repository.impl.UserRepositoryImpl;
+import com.api.social_meli.service.impl.UserServiceImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
 public class UserServiceTest {
+
+    @Mock
+    private UserRepositoryImpl userRepository;
+
+    @InjectMocks
+    private UserServiceImpl userService;
+
+    //region FOLLOW USER
+    @Test
+    @DisplayName("Intentar seguir a usuario existente")
+    void shouldFollowUserSuccessfullyWhenUserExists(){
+        // ARRANGE
+        int userId = 2;
+        int userIdToFollow = 3;
+
+        User user = new User();
+        user.setId(userId);
+        user.setFollowed(new ArrayList<>());
+
+        User userToFollow = new User();
+        userToFollow.setId(userIdToFollow);
+        userToFollow.setFollowers(new ArrayList<>());
+        userToFollow.setPosts(new ArrayList<>(List.of(1)));
```

Comment:
oportunidad de mejora, estos users podrian construirse en MockFactoryUtils para que queden mas cortos los tests. un ejemplo:

```
User user = UserFactory.createUser(userId, List.of(userIdToFollow), new List<User>());
User userToFollow = UserFactory.createUser(userIdToFollow, List.of(userId));
```

Reply:
veo en varios lugares que capaz se puede simplificar el ArrayList, total si no se va a hacer add despues a la lista, podria usarse directo List.of()